weave/trace_server/calls_query_builder.py

```diff
@@ -1281,56 +1297,90 @@ def process_query_to_optimization_sql(
     to avoid filtering out rows where the field is NULL (as they might be part of
     a valid call when combined with other parts).
 
-    Returns an empty string if:
-    1. There are no heavy conditions
-    2. There are OR operations between start-only and end-only fields that cannot be optimized
-
     Performance note: This optimization is critical for queries with heavy fields,
     as it can significantly reduce peak memory by filtering before aggregation.
     """
-    filterable_conditions = [
-        c for c in conditions if c.is_heavy() and not c.is_feedback()
-    ]
-
-    if not filterable_conditions:
+    if not conditions:
         return OptimizationConditions()
 
     def process_operation(
         operation: tsi_query.Operation,
     ) -> Optional[OptimizationConditions]:
         if isinstance(operation, tsi_query.OrOperation):
-            or_conditions = []
+            str_conditions = []
```

Comment:
I think this is fine, but given that you have a second use case now, i cold see a slightly more elegant function here for `process_operation`. Instead of hardcoding the str and datetime conditions, you could instead have a mapper/visitor pattern that allows the caller to specify the conditions to accumulate